.github/workflows/on-pullrequest.yml

```diff
@@ -29,10 +29,55 @@ jobs:
     name: Scan Actions
     uses: ed-fi-alliance-oss/ed-fi-actions/.github/workflows/repository-scanner.yml@main
 
+  upgrade:
+    # Upgrade packages on PR only to avoid a double update when pushed to main
+    name: Upgrade packages
+    if: github.event_name == 'pull_request'
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        working-directory: Meadowlark-js
+    permissions:
+      contents: write
+    steps:
+      - name: Checkout the Repo
+        uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
+        with:
+          fetch-depth: 0w
+
+      - name: Get changes
+        id: changes
+        run: |
+          suggested=v$(cat lerna.json | jq -r .version)
+          current=$(git describe --tags $(git rev-list --tags --max-count=1))
+          echo "Versions: Lerna: $suggested. Current: $current"
+          echo "changes=$([[ "$suggested" != "$current" ]] && echo true)" >>$GITHUB_OUTPUT
+
+      - name: Update versions
+        if: ${{ ! steps.changes.outputs.changes }}
+        run: npx lerna@6.6.2 version prerelease --exact --no-git-tag-version --yes
+
+      - name: Set Version
+        if: ${{ ! steps.changes.outputs.changes }}
+        id: set-version
+        run: |
+          version=v$(cat lerna.json | jq -r .version)
+          echo "version=$version" >> "$GITHUB_OUTPUT"
+
+      - uses: planetscale/ghcommit-action@4131649dbf2fdf1eb34421702972a5af7b0a8731 #v0.1.18
+        if: ${{ ! steps.changes.outputs.changes }}
+        with:
+          commit_message: "${{steps.set-version.outputs.version}}"
+          repo: ${{ github.repository }}
+          branch: ${{ github.head_ref || github.ref_name }}
+        env:
+          GITHUB_TOKEN: ${{secrets.GITHUB_TOKEN}}
+
   analyze-dependencies:
     # This Action will fail if run on anything other than a pull request
     if: github.event_name == 'pull_request'
     name: Analyze Code Dependencies
+    needs: upgrade
```

Comment:
Would it make sense to add a comment explaining why `upgrade` must come first? I'm assuming because, otherwise, the analysis will run against the last version of the packages, instead of the new (current branch) version.